RoaringBitmap/src/main/java/org/roaringbitmap/BitSetUtil.java

```diff
@@ -71,6 +72,71 @@ public static RoaringBitmap bitmapOf(final long[] words) {
     return ans;
   }
 
+  // To avoid memory allocation, reuse ThreadLocal buffers
+  private static final ThreadLocal<long[]> WORD_BLOCK = ThreadLocal.withInitial(() ->
```

Comment:
This is 8 kB that may get allocated for each thread for one function (below) and it never gets released.

I don't think we want that.


What would be accessible is to allow the user to (optionally) pass a buffer to the function below. If the user passes a buffer, then you use it, otherwise, you allocate it. 

In this manner, you give the use full control over the performance, and you don't make people pay 8 kB that they don't want to lock down for one function.